timeutil/schedule.go

```diff
@@ -708,3 +708,47 @@ func parseEventSet(s string) (*Schedule, error) {
 
 	return &schedule, nil
 }
+
+// Includes checks whether given time t falls inside the time range covered by
+// the schedule. A single time schedule eg. '10:00' is treated as spanning the
+// time <10:00, 10:01)
+func (sched *Schedule) Includes(t time.Time) bool {
+	if len(sched.WeekSpans) > 0 {
+		var weekMatch bool
+		for _, week := range sched.WeekSpans {
+			if week.Match(t) {
+				weekMatch = true
+				break
+			}
+		}
+		if !weekMatch {
+			return false
+		}
+	}
+
+	for _, tspan := range sched.flattenedClockSpans() {
+		window := tspan.Window(t)
+		if window.End.Equal(window.Start) {
+			// schedule granularity is a minute, a schedule '10:00'
+			// in fact is: <10:00, 10:01)
+			window.End = window.End.Add(time.Minute)
```

Comment:
Should we make sure tspan.Window(..) already applies this logic internally? If window always needs to be >=1 minute, then I think this should be applied whenever windows is constructed.

Reply:
I think it may be ok to update `ClockSpan.Window()` to handle this. @niemeyer what's your take on this? If we have a schedule `12:00`, does that mean there is a window `<12:00, 12:01)` ?

Reply:
> Should we make sure tspan.Window(..) already applies this logic internally? If window always needs to be >=1 minute, then I think this should be applied whenever windows is constructed.

Keeping it inside _Includes_ as done now seems like the right thing to do, as this is an implementation detail and can change over time. Otherwise we mutate the real input that was parsed, and then we cannot distinguish a range that was actually provided as 12:00-12:01 from one provided as simply 12:00 or 12:00-12:00, which in turn means we'll lose the ability to roundtrip the representation (String won't report what was parsed), and we'll also lose the ability to adapt the implementation into the future for data that was stored after parsing (say, change the window to 3 minutes instead of 1).